WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java

```diff
@@ -157,9 +163,21 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
                                      String.valueOf(AppPrefs.getVideoOptimizeQuality()),
                                      getLabelForVideoEncoderBitrateValue(AppPrefs.getVideoOptimizeQuality()));
 
+        mEnableGutenberg.setChecked(AppPrefs.isGutenbergEditorEnabled());
+
+        if (BuildConfig.FLAVOR.equals("vanilla")) {
```

Comment:
I think it would probably be more clear if we introduce a feature flag which gets defined in gradle, in the product flavors section. For example, see the `buildConfigField`s in [here](https://github.com/wordpress-mobile/WordPress-Android/blob/191e795d58d4273ed2778aa02c0206d338acab82/WordPress/build.gradle#L37-L66)

That way, it's easier to understand which flavor enabled which feature just by looking at the gradle file.

Then, in the code, we'll check for the feature flag value instead of the flavor type.

What do you think?

Reply:
@hypest I think that you have proposed a good solution, agree. Let's change it.

Reply:
@hypest please take another look. Fixed.